### PR TITLE
update pilot mesh config default

### PR DESCRIPTION
**Pull request by linsun**
to sync with what is shipped in the /etc/istio/config/mesh file.  fixes:  https://github.com/istio/istio/issues/11909

### CODE REVIEW

File: pilot/pkg/model/context.go [hunk 1]
```diff
@@ -401,20 +401,23 @@ func DefaultMeshConfig() meshconfig.MeshConfig {
...
 		MixerCheckServer:                  "",
 		MixerReportServer:                 "",
-		DisablePolicyChecks:               false,
+		DisablePolicyChecks:               true,
 		PolicyCheckFailOpen:               false,
 		SidecarToTelemetrySessionAffinity: false,
 		ProxyListenPort:                   15001,
 		ConnectTimeout:                    types.DurationProto(1 * time.Second),
-		IngressClass:                      "istio",
 		IngressControllerMode:             meshconfig.MeshConfig_STRICT,
```
Comment (rshriram) [inline pilot/pkg/model/context.go]: i think this should be disabled instead of enable.. isn't it?

File: pilot/pkg/model/context.go [hunk 2]
```diff
@@ -381,8 +381,8 @@ func DefaultProxyConfig() meshconfig.ProxyConfig {
 		ConfigPath:             ConfigPathDir,
 		BinaryPath:             BinaryPathFilename,
 		ServiceCluster:         ServiceClusterName,
-		DrainDuration:          types.DurationProto(2 * time.Second),
-		ParentShutdownDuration: types.DurationProto(3 * time.Second),
+		DrainDuration:          types.DurationProto(45 * time.Second),
+		ParentShutdownDuration: types.DurationProto(60 * time.Second),
```
Comment (rshriram) [inline pilot/pkg/model/context.go]: undo this please..we set this to 2/3 for a reason, after lots of experimentation

File: pilot/pkg/model/context.go [hunk 3]
```diff
@@ -363,7 +363,7 @@ const (
 	ServiceClusterName = "istio-proxy"
 
 	// DiscoveryPlainAddress discovery IP address:port with plain text
-	DiscoveryPlainAddress = "istio-pilot:15007"
+	DiscoveryPlainAddress = "istio-pilot:15010"
```
Comment (rshriram) [inline pilot/pkg/model/context.go]: ??

**Comment by linsun** [comment]
@rshriram fyi, this is our default /etc/istio/config/mesh file mounted to pilot pod.  I'm not proposing any changes, just to make them sync.

```
# Set the following variable to true to disable policy checks by the Mixer.
# Note that metrics will still be reported to the Mixer.
disablePolicyChecks: true

# Set enableTracing to false to disable request tracing.
enableTracing: true

# Set accessLogFile to empty string to disable access log.
accessLogFile: "/dev/stdout"

# If accessLogEncoding is TEXT, value will be used directly as the log format
# example: "[%START_TIME%] %REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%\n"
# If AccessLogEncoding is JSON, value will be parsed as map[string]string
# example: '{"start_time": "%START_TIME%", "req_method": "%REQ(:METHOD)%"}'
# Leave empty to use default log format
accessLogFormat: ""

# Set accessLogEncoding to JSON or TEXT to configure sidecar access log
accessLogEncoding: 'TEXT'
#
# Deprecated: mixer is using EDS
# Let Pilot give ingresses the public IP of the Istio ingressgateway
ingressService: istio-ingressgateway

# Unix Domain Socket through which envoy communicates with NodeAgent SDS to get
# key/cert for mTLS. Use secret-mount files instead of SDS if set to empty. 
sdsUdsPath: 

# This flag is used by secret discovery service(SDS). 
# If set to true(prerequisite: https://kubernetes.io/docs/concepts/storage/volumes/#projected), Istio will inject volumes mount 
# for k8s service account JWT, so that K8s API server mounts k8s service account JWT to envoy container, which 
# will be used to generate key/cert eventually. This isn't supported for non-k8s case.
enableSdsTokenMount: false

# This flag is used by secret discovery service(SDS). 
# If set to true, envoy will fetch normal k8s service account JWT from '/var/run/secrets/kubernetes.io/serviceaccount/token' 
# (https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod) 
# and pass to sds server, which will be used to request key/cert eventually. 
# this flag is ignored if enableSdsTokenMount is set.
# This isn't supported for non-k8s case.
sdsUseK8sSaJwt: false

# The trust domain corresponds to the trust root of a system.
# Refer to https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain
trustDomain: 

# Set the default behavior of the sidecar for handling outbound traffic from the application:
# REGISTRY_ONLY - restrict outbound traffic to services defined in the service registry as well
#   as those defined through ServiceEntries
# ALLOW_ANY - outbound traffic to unknown destinations will be allowed, in case there are no
#   services or ServiceEntries for the destination port
outboundTrafficPolicy:
  mode: REGISTRY_ONLY
defaultServiceExportTo:
  - "*"
defaultVirtualServiceExportTo:
  - "*"
defaultDestinationRuleExportTo:
  - "*"

defaultConfig:
  #
  # TCP connection timeout between Envoy & the application, and between Envoys.
  connectTimeout: 10s
  #
  ### ADVANCED SETTINGS #############
  # Where should envoy's configuration be stored in the istio-proxy container
  configPath: "/etc/istio/proxy"
  binaryPath: "/usr/local/bin/envoy"
  # The pseudo service name used for Envoy.
  serviceCluster: istio-proxy
  # These settings that determine how long an old Envoy
  # process should be kept alive after an occasional reload.
  drainDuration: 45s
  parentShutdownDuration: 1m0s
  #
  # The mode used to redirect inbound connections to Envoy. This setting
  # has no effect on outbound traffic: iptables REDIRECT is always used for
  # outbound connections.
  # If "REDIRECT", use iptables REDIRECT to NAT and redirect to Envoy.
  # The "REDIRECT" mode loses source addresses during redirection.
  # If "TPROXY", use iptables TPROXY to redirect to Envoy.
  # The "TPROXY" mode preserves both the source and destination IP
  # addresses and ports, so that they can be used for advanced filtering
  # and manipulation.
  # The "TPROXY" mode also configures the sidecar to run with the
  # CAP_NET_ADMIN capability, which is required to use TPROXY.
  #interceptionMode: REDIRECT
  #
  # Port where Envoy listens (on local host) for admin commands
  # You can exec into the istio-proxy container in a pod and
  # curl the admin port (curl http://localhost:15000/) to obtain
  # diagnostic information from Envoy. See
  # https://lyft.github.io/envoy/docs/operations/admin.html
  # for more details
  proxyAdminPort: 15000
  #
  # Set concurrency to a specific number to control the number of Proxy worker threads.
  # If set to 0 (default), then start worker thread for each CPU thread/core.
  concurrency: 0
  #
  tracing:
    zipkin:
      # Address of the Zipkin collector
      address: zipkin.istio-system:9411
  #
  # Mutual TLS authentication between sidecars and istio control plane.
  controlPlaneAuthPolicy: NONE
  #
  # Address where istio Pilot service is running
  discoveryAddress: istio-pilot.istio-system:15010```

File: pilot/pkg/model/context.go (hunk 1 above)
Comment (linsun) [reply pilot/pkg/model/context.go]: Yes the ingressClass is gone now in our helm/istio/templates/configmap.yaml

File: pilot/pkg/model/context.go (hunk 2 above)
Comment (linsun) [reply pilot/pkg/model/context.go]: this is the default set in our helm/istio/templates/configmap.yaml

File: pilot/pkg/model/context.go (hunk 3 above)
Comment (linsun) [reply pilot/pkg/model/context.go]: also the default we use in our helm/istio/templates/configmap.yaml

**Comment by linsun** [comment]
@wenchenglu can this be merged in 1.1?

**Pull request merged**